src/server/scripts/Northrend/Ulduar/Ulduar/boss_assembly_of_iron.cpp

```diff
@@ -626,11 +629,20 @@ class boss_stormcaller_brundir : public CreatureScript
                             {
                                 float x = float(irand(-25, 25));
                                 float y = float(irand(-25, 25));
-                                me->GetMotionMaster()->MovePoint(0, me->GetPositionX() + x, me->GetPositionY() + y, me->GetPositionZ());
+
+                                Position pos = me->GetPosition();
+
+                                pos.m_positionX += x;
+                                pos.m_positionY += y;
+
                                 // Prevention to go outside the room or into the walls
-                                if (Creature* trigger = me->FindNearestCreature(NPC_WORLD_TRIGGER, 100.0f, true))
-                                    if (me->GetDistance(trigger) >= 50.0f)
+                                if (auto trigger = ObjectAccessor::GetCreature(*me, m_TriggerGUID))
```

Comment:
Do not use `auto` for anything except container iterators (or other super long deep nested types)